### DISCUSSION TITLE
How to set basemap's maximum pitch?

### DISCUSSION THREAD

**Discussion by yxia-2022** [visgl/deck.gl#7286] [Q&A]
I can set the maxPitch of the elements except from the basemap by

```
initialViewState: {
  pitch: ...,
  maxPitch: 90,
  ...
},
```

However, the basemap still has a maximum pitch at about 60 deg. The below is an example. I set the maxPitch of the `initialViewState` to 90, and 

- At 0 deg pitch

<img width="999" alt="image" src="https://user-images.githubusercontent.com/111309004/191173203-ad322d56-9bb7-42d3-9806-0f081b405917.png">

- At close to 90 deg pitch

<img width="1329" alt="image" src="https://user-images.githubusercontent.com/111309004/191173374-6fec19fd-0db0-4e16-b79e-c335f0c27ab2.png">

So only the lines are rotated accordingly. I've tried my best to search for a solution, and it seems that if there's any way to increase the basemap max pitch, it should be inside `mapStyle: 'https://basemaps.cartocdn.com/gl/dark-matter-nolabels-gl-style/style.json'`. Unfortunately, I didn't find any related setting there.

**Comment by Pessimistress** [comment]
There is a `maxPitch` option in mapbox-gl / react-map-gl.

**Reply by yxia-2022** [reply]
@Pessimistress Thank you so much! Let me try it out.

**Comment by yxia-2022** [comment]
Hi, I'm trying out pydeck, and I can pass `max_pitch` to `pdk.ViewState`, and my view type is "MapView". The map itself will have 60 deg pitch in maximum, regardless that `max_pitch` can be 90. Would it be possible to assign the `max_pitch` to the map layer in pydeck as well?
